### PR TITLE
Add peak [task] total memory to returned stats 

### PR DESCRIPTION
**Pull request by aweisberg**
```
== RELEASE NOTES ==

General Changes
* Add peak [task] total memory to returned stats 
```

### CODE REVIEW

**Comment by aweisberg** [comment]
@caithagoras only including you to look at https://github.com/prestodb/presto/pull/14324/files#diff-193de7b76fc87b40f406fcba9d02e920R35

Basically in the near future the clusters will enforce memory limits based on total memory limit not just user memory. So verifier will probably need to change how it automatically resolves differences in limits.

**Comment by caithagoras** [comment]
@aweisberg That part looks good to me.

So, correct me if I'm wrong.
- peak memory is user only.
- peak total memory is user + system.
- What about peak task total?

We're currently enforcing limit on "peak memory" and will instead enforce on "peak total memory"?

**Comment by aweisberg** [comment]
You have it correctly. We will also enforce based on peak task total instead of peak task.

I think this will work fine for verifier because when we run verifier we will have the same limits when testing each version of Presto.

File: presto-cli/src/main/java/com/facebook/presto/cli/StatusPrinter.java [hunk 1]
```diff
@@ -206,6 +206,10 @@ public void printFinalInfo()
 
             // Peak Memory: 1.97GB
             reprintLine("Peak Memory: " + formatDataSize(bytes(stats.getPeakMemoryBytes()), true));
```
[the remark this answers is not in the document]
Comment (aweisberg) [reply presto-cli/src/main/java/com/facebook/presto/cli/StatusPrinter.java:208]: I agree our terminology is not great, but it is our terminology. We don't refer to user memory in Daiquery, Chronos, Presto Query Lookup, Presto Query Status, or in the code most of the time.

WDYT?

[the remark this answers is not in the document]
Comment (rschlussel) [reply presto-cli/src/main/java/com/facebook/presto/cli/StatusPrinter.java:208]: Those are internal tools and shouldn't affect what Presto surfaces.

That being said, I think we're a bit inconsistent, in our terminology, as the code does in places refer to "peakUserMemoryBytes".  I think that's less confusing and we should try make anything user facing start to use that terminology if we're going to be surfacing the "total memory" more. To me, Peak Memory and Peak Total Memory sound like they should mean exactly the same thing and it's confusing that they don't.  Peak user memory and peak total memory at least sound like they might mean different things.  Even if you don't know what user memory is, you'd probably infer that there's some other kind of memory that total memory includes. 

Also, possibly controversial opinion (for a bigger discussion separate from this PR), but I don't really think separating user memory and system/total memory is that valuable.  I mean, it gives a quick way to see whether a query is using lots of memory from reading maps or if it's from building hash tables and sorting, but you could also learn that from the operator stats. From a cluster management perspective it's not relevant, and for a user just trying to understand the resource utilization of their query, I think it's more confusing than helpful.

[the remark this answers is not in the document]
Comment (aweisberg) [reply presto-cli/src/main/java/com/facebook/presto/cli/StatusPrinter.java]: I agree it's a distinction that doesn't seem useful most of the time.

I'll change it. I misunderstood what you were asking for anyways. My brain added that you wanted to switch total to System along with User.

**Pull request merged**